main.go

```diff
@@ -116,13 +117,11 @@ func main() {
 		}
 	}()
 
-	clusterVersion, err := getClusterVersion(mgr.GetAPIReader())
+	info, err := clusterinfo.Get(context.Background(), mgr.GetAPIReader(), getOpenshiftNS())
```

Comment:
Why do we need a NS here?  If it is for HCP, then this should be the NS of the CLF which means you need to fetch this on every reconcile or cache entries somewhere.  If this is for nonHCP, clusterinfo AFAIK is a clusterwide resource which has no need of a NS.

Reply:
You are right - This is for HCP, we need to get it on every reconcile and cache it so it can be used for collector generation for that reconcile. I need to rethink this bit...